rdrf/rdrf/admin.py

```diff
@@ -130,104 +174,142 @@ def export_registry(registry, request):
-    for registry in registry_models_selected:
-        registry.generate_questionnaire()
+            existing_notifications = EmailNotification.objects.filter(
+                registry=registry,
+                description__in=EventType.REGISTRATION_TYPES
+            )
 
+            if len(existing_notifications) == 0:
+                messages.warning(request, f"Notifications need to be created for registry '{registry.code}'. "
+                                          f"Use the 'Create notifications' action")
 
-generate_questionnaire_action.short_description = _("Generate Questionnaire")
+        # Check for missing cache table
+        for cache_alias in settings.CACHES:
+            cache = caches[cache_alias]
+            if hasattr(cache, '_table') and cache._table not in connection.introspection.table_names():
+                messages.error(request, _(f"Missing cache table '{cache._table}'. Run django-admin createcachetable"))
 
+        if len(unchanged) > 0:
+            messages.info(request, f"'{', '.join(code for code in unchanged)}' already enabled registration")
 
-class RegistryAdmin(admin.ModelAdmin):
-    actions = [export_registry_action, design_registry_action, generate_questionnaire_action]
+    enable_registration_action.short_description = _("Enable registration")
 
-    def get_queryset(self, request):
-        if not request.user.is_superuser:
-            user = get_user_model().objects.get(username=request.user)
-            return Registry.objects.filter(registry__in=[reg.id for reg in user.registry.all()])
+    def disable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
-        return Registry.objects.all()
+        for registry in registry_models_selected:
+            if not registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(registry.code)
 
-    def has_add_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+            registry.remove_feature(RegistryFeatures.REGISTRATION)
+            registry.save()
 
-    def has_change_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+            messages.success(request, f"Registration disabled for '{registry.code}'")
 
-    def has_delete_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+        if len(unchanged) > 0:
+            messages.info(request, f"'{', '.join(code for code in unchanged)}' already had registration disabled")
 
-    def get_urls(self):
-        original_urls = super(RegistryAdmin, self).get_urls()
+    disable_registration_action.short_description = _("Disable registration")
 
-        return original_urls
+    def create_notifications_action(self, request, registry_models_selected):
```

Comment:
I am not sure about this admin action.
There are multiple types of notifications, so it isn't really clear what it would do.
I think we shouldn't have a create notifications admin action yet.

We could start by having business logic code to create default registration notifications and we could call it directly from the "Enable registration" action above.

Reply:
See my later comment